ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -343,10 +382,22 @@ private CloudBlobContainer getContainer(BlobId blobId, boolean autoCreate)
     String partitionPath = "partition-" + blobId.getPartition().toPathString();
     CloudBlobContainer azureContainer = azureBlobClient.getContainerReference(partitionPath);
     if (autoCreate) {
+      ensureCreated(azureContainer);
+    }
+    return azureContainer;
+  }
+
+  /**
+   * Utility method to ensure that the requested container exists in the storage account.
+   * @param azureContainer the container that must exist.
+   * @throws StorageException if the operation fails.
+   */
+  private void ensureCreated(CloudBlobContainer azureContainer) throws StorageException {
+    if (!knownContainers.contains(azureContainer.getName())) {
       azureContainer.createIfNotExists(BlobContainerPublicAccessType.CONTAINER, new BlobRequestOptions(),
```

Comment:
So we're not worried about containers getting deleted at this point?

Reply:
TBH I hadn't considered that, but thinking about it now, if someone deletes the container it will likely be part of wiping out the backup (at least for that partition) and starting over, in which case the VCR should be restarted.  And if the container is deleted accidentally or maliciously, IMO it will be better for the VCR to start throwing exceptions and up the error count rather than silently recreating the container and marching on.  So I prefer to keep the code as it is.